hidimstat/estimator/_utils/u_Dnn_learner.py

```diff
@@ -0,0 +1,657 @@
+                list_stacking[ind_layer] = layer(x[:, self.list_grps[ind_layer]])
+            x = torch.cat(list_stacking, dim=1)
+        return self.layers(x)
+
+    def training_step(self, batch, device, problem_type):
+        X, y = batch[0].to(device), batch[1].to(device)
+        y_pred = self(X)  # Generate predictions
+        if problem_type == "regression":
+            loss = F.mse_loss(y_pred, y)
+        elif problem_type == "classification":
+            loss = F.cross_entropy(y_pred, y)  # Calculate loss
+        else:
+            loss = F.binary_cross_entropy_with_logits(y_pred, y)
+        return loss
+
+    def validation_step(self, batch, device, problem_type):
+        X, y = batch[0].to(device), batch[1].to(device)
+        y_pred = self(X)  # Generate predictions
+        if problem_type == "regression":
+            loss = F.mse_loss(y_pred, y)
+            return {
+                "val_mse": loss,
+                "batch_size": len(X),
+            }
+        else:
+            if problem_type == "classification":
+                loss = F.cross_entropy(y_pred, y)  # Calculate loss
+            else:
+                loss = F.binary_cross_entropy_with_logits(y_pred, y)
+            acc = self.accuracy(y_pred, y.int())
+            return {
+                "val_loss": loss,
+                "val_acc": acc,
+                "batch_size": len(X),
+            }
+
+    def validation_epoch_end(self, outputs, problem_type):
+        if problem_type in ("classification", "binary", "ordinal"):
+            batch_losses = []
+            batch_accs = []
+            batch_sizes = []
+            for x in outputs:
+                batch_losses.append(x["val_loss"] * x["batch_size"])
+                batch_accs.append(x["val_acc"] * x["batch_size"])
+                batch_sizes.append(x["batch_size"])
+            self.loss = torch.stack(batch_losses).sum().item() / np.sum(
+                batch_sizes
+            )  # Combine losses
+            epoch_acc = torch.stack(batch_accs).sum().item() / np.sum(
+                batch_sizes
+            )  # Combine accuracies
+            return {"val_loss": self.loss, "val_acc": epoch_acc}
+        else:
+            batch_losses = [x["val_mse"] * x["batch_size"] for x in outputs]
+            batch_sizes = [x["batch_size"] for x in outputs]
+            self.loss = torch.stack(batch_losses).sum().item() / np.sum(
+                batch_sizes
+            )  # Combine losses
+            return {"val_mse": self.loss}
+
+    def epoch_end(self, epoch, result):
+        if len(result) == 2:
+            print(
+                "Epoch [{}], val_loss: {:.4f}, val_acc: {:.4f}".format(
+                    epoch + 1, result["val_loss"], result["val_acc"]
+                )
+            )
+        else:
+            print("Epoch [{}], val_mse: {:.4f}".format(epoch + 1, result["val_mse"]))
+
+
+def _evaluate(model, loader, device, problem_type):
```

Comment:
docstring (1 line)

Reply:
This will be addressed by the issue #81.